app/src/main/java/vn/mbm/phimp/me/Camera2.java

```diff
@@ -98,200 +121,211 @@ public void onOrientationChanged(int orientation) {
+					}else
 						if (statusScreen == 1){
-						statusScreen = 0;					
+						statusScreen = 0;
 						if (PhimpMe.flashStatus == 0) flash.setImageDrawable(getResources().getDrawable(R.drawable.thunder_gray_r));
 						else if (PhimpMe.flashStatus == 1) flash.setImageDrawable(getResources().getDrawable(R.drawable.thunder_r));
 						else flash.setImageDrawable(getResources().getDrawable(R.drawable.thunder_a_r));
 						buttonClick.setImageDrawable(getResources().getDrawable(R.drawable.icon_camera_r));
-						camera_switch.setImageDrawable(getResources().getDrawable(R.drawable.camera_32_r));											
+						camera_switch.setImageDrawable(getResources().getDrawable(R.drawable.camera_32_r));
 				}*/
 			}
 		};
-		mOrientation.enable();		
+		mOrientation.enable();
 		Log.d(TAG, "onCreate'd");
 		//PhimpMe.hideTabs();
-	}	
-	public void setLayout(int layout){
-		setContentView(layout);
-		frame = ((FrameLayout) findViewById(R.id.preview));	   
-		preview = new Preview(this);
+	}
+
+	public void setLayout() {
+//		setContentView(layout);
+		frame = ((FrameLayout) view.findViewById(R.id.preview));
+		preview = new Preview(getActivity());
 		//setContentView(preview);
 		locationListener = new LocationListener() {
 
-            public void onStatusChanged(String provider, int status, Bundle extras) {
-
-            }
-
-            public void onProviderEnabled(String provider) {
+			public void onStatusChanged(String provider, int status, Bundle extras) {
 
-            }
+			}
 
-            public void onProviderDisabled(String provider) {
+			public void onProviderEnabled(String provider) {
 
-            }
+			}
 
-            public void onLocationChanged(Location location) {
+			public void onProviderDisabled(String provider) {
 
-               // Camera2.this.gpsLocationReceived(location);
-                lat = locationManager.getLastKnownLocation(LocationManager.NETWORK_PROVIDER).getLatitude();
-                lon = locationManager.getLastKnownLocation(LocationManager.NETWORK_PROVIDER).getLongitude();
-            }
+			}
 
-        };
-        locationManager = (LocationManager)getSystemService(Context.LOCATION_SERVICE);
-        Criteria locationCritera = new Criteria();
-        locationCritera.setAccuracy(Criteria.ACCURACY_COARSE);
-        locationCritera.setAltitudeRequired(false);
-        locationCritera.setBearingRequired(false);
-        locationCritera.setCostAllowed(true);
-        locationCritera.setPowerRequirement(Criteria.NO_REQUIREMENT);
-        String providerName = locationManager.getBestProvider(locationCritera, true);
+			public void onLocationChanged(Location location) {
+
+				// Camera2.this.gpsLocationReceived(location);
+				if (ActivityCompat.checkSelfPermission(getActivity(), Manifest.permission.ACCESS_FINE_LOCATION) != PackageManager.PERMISSION_GRANTED && ActivityCompat.checkSelfPermission(getContext(), Manifest.permission.ACCESS_COARSE_LOCATION) != PackageManager.PERMISSION_GRANTED) {
```

Comment:
The app crashing on this line.
Here is the log: 
```
java.lang.NullPointerException: Attempt to invoke virtual method 'int android.content.Context.checkPermission(java.lang.String, int, int)' on a null object reference
                                                                     at android.support.v4.content.ContextCompat.checkSelfPermission(ContextCompat.java:453)
                                                                     at vn.mbm.phimp.me.Camera2$2.onLocationChanged(Camera2.java:171)
                                                                     at android.location.LocationManager$ListenerTransport._handleMessage(LocationManager.java:281)
                                                                     at android.location.LocationManager$ListenerTransport.access$000(LocationManager.java:210)
                                                                     at android.location.LocationManager$ListenerTransport$1.handleMessage(LocationManager.java:226)
                                                                     at android.os.Handler.dispatchMessage(Handler.java:102)
                                                                     at android.os.Looper.loop(Looper.java:135)
                                                                     at android.app.ActivityThread.main(ActivityThread.java:5268)
                                                                     at java.lang.reflect.Method.invoke(Native Method)
                                                                     at java.lang.reflect.Method.invoke(Method.java:372)
                                                                     at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:902)
                                                                     at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:697)
                                                                 
                                                                 
                                                                 --------- beginning of main
```
This might be because of Context in checkSelfPermission. Look in to it and update code. :+1: